src/tactic/core.lean

```diff
@@ -497,17 +497,37 @@ meta def revert_deps (e : expr) : tactic ℕ := do
 meta def is_local_def (e : expr) : tactic unit :=
 retrieve $ do revert e, expr.elet _ _ _ _ ← target, skip
 
+private def partition_local_deps_aux {α} [decidable_eq α] (vs : list α) : list α → list α → list (list α)
+| [] acc := [acc.reverse]
+| (l :: ls) acc :=
+  if l ∈ vs then acc.reverse :: partition_local_deps_aux ls [l]
+  else partition_local_deps_aux ls (l :: acc)
+
+/-- `partition_local_deps vs`, with `vs` a list of local constants,
+reorders `vs` in the order they appear in the local context together
+with the variables that follow them. If local context is `[a,b,c,d,e,f]`,
+and that we call `partition_local_deps [d,b]`, we get `[[b,c],[d,e,f]]`.
+The head of each list is one of the variables given as a parameter. -/
+meta def partition_local_deps (vs : list expr) : tactic (list (list expr)) :=
+do ls ← local_context,
+   pure (partition_local_deps_aux vs ls []).tail.reverse
```

Comment:
Can you use `list.​split_on_p` instead of `partition_local_deps_aux` here?

Reply:
It differs from `split_on_p` because when `partition_local_deps_aux` finds a match, it starts a new list but it adds the match to that list.

Reply:
It seems like for this use cases something like `split_on_p'` with output `list (sum (list T) T)`, so that
`split_on_p is_even [1, 1, 2, 3, 4, 4]` would give `[1, 1], 2, [3], 4, 4`

Reply:
@eric-wieser I'm not sold. I think it would work but I don't think it would be simpler than what I have because it is not a drop in replacement and that it is not clear how generally useful the combinator is.

Reply:
That's fine - more a question of whether that would be a useful addition to list in general.

Reply:
Ok, then please add a doc-string to `partition_local_deps_aux` with the difference with `split_on_p`.